shared_model/builders/protobuf/builder_templates/query_template.hpp

```diff
@@ -204,6 +204,13 @@ namespace shared_model {
 
       auto build() const {
         static_assert(S == (1 << TOTAL) - 1, "Required fields are not set");
+        if (not query_.has_payload()) {
+          throw std::invalid_argument("Query missing payload");
```

Comment:
Do we catch this exception in all places where we use TemplateQueryBuilder?

Reply:
Yes, it is mapped to optional in server code, and caught manually by client library users.